### DISCUSSION TITLE
Creating a unique thumbnail for each clip

### DISCUSSION THREAD

**Discussion by oxmophoto** [Breakthrough/DVR-Scan#152] [Q&A]
I am a motorsports photographer/videographer and am hoping that DVR-Scan will shave a few hours (days, probably in some instances) off of what I am planning on doing. I was wondering if anyone had any ideas as to how to make the thumbnail of the exported video clips be the middle of each clip and not the beginning.

I was hoping to make it easy for the riders at a school I work with to find clips of themselves as they go through a section of track without having to click on each video. As it is now, every clip looks exactly the same (blank race track) until they enter the frame.

I have literally zero coding experience (it took me about an hour to figure out what I was doing with already provided code) so my apologies if this is either something that I should have know was impossible or should have known how to figure out.

Thanks in advance for any assistance.

**Comment by Breakthrough** [comment]
When you say thumbnails, are you referring to the preview that shows up in a file explorer?  DVR-Scan doesn't have control over what frame your operating system might be displaying when browsing through video clips unfortunately.  I don't think most OSes allow overriding this either...

That being said, you might want to consider creating a separate *image* with the same name of the video. You can use something like `ffmpeg` to save an image after X seconds on the resulting videos and save those images with each clip.

For example to save a frame from a video at the 4 second mark:

    ffmpeg -ss 00:00:04 -i input.mp4 -frames:v 1 output.jpg

Happy to help out and point you in the right direction if you can't get the above working.  You can modify this to run on all videos in a directory for example.

**Reply by oxmophoto** [reply]
Yes, I should have been more clear there. Thank you for the quick response.

I wasn't sure if DVR-Scan had control of what I was seeing after the file creation. I appreciate the additional code idea. I do have the ability to select a thumbnail image once the videos are uploaded to my smugmug site, but given that I am going to have a few hundred clips at each event, this turns into a bunch of time after upload. 

In a perfect world I wouldn't have to do any additional work after the main file was divided up, but it looks like I still live in a reality-based situation.

**Reply by Breakthrough** [reply]
That's unfortunate... SmugMug [does have a documented API](https://api.smugmug.com/api/v2/doc/index.html) but so you might be able to use that + wget or a Python script to handle this more automatically (e.g. upload a video + set the thumbnail to an image with the same name).

As for generating the thumbnails, depending on the OS you're using, you might be able to just use wildcards in the file paths to do entire folders of clips.

Thanks for sharing your use case by the way - DVR-Scan has no telemetry, so experiences like yours are valuable as they help me understand how most people use the software.

**Reply by oxmophoto** [reply]
I am on windows, and will look into your suggestions. As I mentioned I am a coding idiot, (wasn't good at it in the late 80's, not any better now) but I have some more intrepid friends that may help me muddle through my inadequacies. Thanks again for your assistance, and for coding up this bit of magic. Not having to sift through minutes of "dead air" as I waif for riders to finish a 2-5 minute lap is a step in the right direction for me to save a ton of time this season as I try to expand more into the video side of things.
